client/src/containers/NotRegisteredPage.tsx

```diff
@@ -175,13 +175,32 @@ export default class NotRegisteredPage extends Component<Props, State> {
           <div className="HomePage__content">
             <div className="HomePage__search">
               <h5 className="mt-10 text-danger text-center text-bold text-large">
-                {usersInputAddress ? (
-                  <Trans>No registration found for {usersInputAddressFragment}!</Trans>
-                ) : (
-                  <Trans>No registration found!</Trans>
-                )}
+                {registrationMessage}
               </h5>
-              {buildingTypeMessage}
+              {buildingInfo.lastregistrationdate ? (
+                <p className="text-center">
+                  <Trans>
+                    The registration for this property is missing details for owner name or
+                    businesses address, which are required to link the property to a portfolio.
+                  </Trans>
+                </p>
+              ) : (
+                <></>
+              )}
+              {buildingInfo.unitsres > 0 && registrationMissingOrExpired ? (
+                <>
+                  {buildingTypeMessage}
+                  {lastRegisteredCard}
+                  {failedToRegisterLink}
+                </>
+              ) : (
+                <></>
+              )}
+              {buildingInfo.unitsres === 0 && registrationMissingOrExpired ? (
+                <>{buildingTypeMessage}</>
+              ) : (
+                <></>
+              )}
```

Comment:
I wonder if this logic can be condensed a little bit. Since both cases show the `buildingTypeMessage`, I'm wondering if you keep these as the same case and just have `{lastRegisteredCard}` and `{failedToRegisterLink}` render depending on the number of units? This is another style question so it's your call and I could see arguments either way.

Reply:
Yeah, I wasn't really happy with this either. I looked through some other places and realized you can use `boolean && (<jsx>)` and not everything needs to be a ternary, so I made some more changes to hopefully make this whole section more clear and readable, and also reduce some repetition elsewhere.